soroban-env-host/src/test/token.rs

```diff
@@ -991,6 +991,56 @@ fn test_clawback_on_contract() {
     assert_eq!(token.balance(user_2_addr).unwrap(), 100_000_000);
 }
 
+#[test]
+fn test_auth_revocable_on_contract() {
+    let test = TokenTest::setup();
+    let admin = TestSigner::account(&test.issuer_key);
+    let token = test.default_token();
+
+    let issuer_ledger_key = test
+        .host
+        .to_account_key(keypair_to_account_id(&test.issuer_key));
+
+    let user_1 = generate_bytes_array();
+    let user_1_addr = contract_id_to_address(&test.host, user_1);
+
+    // contract is authorized by default
+    assert!(token.authorized(user_1_addr.clone()).unwrap());
+
+    // clear all flags
+    test.update_account_flags(&issuer_ledger_key, 0);
+
+    // can't deauthorize because issuer does not have auth revocable set
+    assert_eq!(
+        to_contract_err(
+            token
+                .set_authorized(&admin, user_1_addr.clone(), false)
+                .err()
+                .unwrap()
+        ),
+        ContractError::OperationNotSupportedError
+    );
+
+    // set auth revocable
+    test.update_account_flags(&issuer_ledger_key, AccountFlags::RevocableFlag as u32);
+
+    // Now auth can be revoked
+    token
+        .set_authorized(&admin, user_1_addr.clone(), false)
+        .unwrap();
+
+    assert!(!token.authorized(user_1_addr.clone()).unwrap());
```

Comment:
Do we have coverage for making sure that ops actually don't succeed?

Reply:
I don't think I understand your question.

Reply:
Do we have a test that does `token.transfer(user_1_addr,...)` when `user_1_addr` is a deauthorized contract ?

Reply:
It looks like we're missing that for contract balances (I think we're missing quite a few test cases for contract balances actually).

Reply:
Right, so can we add it?

Reply:
Ah I see what you originally meant. I have a list of tests we need to add, and would prefer to do this when we address the missing tests (I need to open a github issue with that list.)

Reply:
Ok, SG.